test/extensions/coin-machine.js

```diff
@@ -537,7 +537,7 @@ contract("Coin Machine", (accounts) => {
       expect(currentPrice).to.eq.BN(WAD);
 
       // Now we reload the token balance (and advance a period)
-      // Price doesn't adjust in the period you update
+      // Price adjusts in the period you update
```

Comment:
Is this a change in behaviour that we're happy with? Was a requirement along these lines how the issue crept in in the first place - a guard against an unintended drop in price (because of selling a number of tokens below the target, but only because we ran out)?

Reply:
> I would default to not changing existing behaviour wherever possible

While this is true in general, I also I _really_ don't want to have to update [the logic in the dapp](https://github.com/JoinColony/colonyDapp/blob/3699bb1424c2eeb88c0319f5fa2c512df1019de1/src/data/resolvers/coinMachine.ts).

Reply:
It seems more fair in general for both token seller and token buyer to stop price evolution as soon as buys are not able to happen.

Regarding having to update the resolver logic for the DApp, this is likely going to need to be revised anyway, to move it over to the new CDApp architecture which will also be used for CoinMachine.

Reply:
I had a conversation with Arren on Discord to explain the issue - let me know if you feel like I've misrepresented something (as I said to you on Discord, I didn't follow the whole of your message):

**Alex**
> I think the way to think about it is (though at this point, I'm not 100% certain)
>
> Period n: We sell out of tokens, evolution stops, purchasing is not possible
>_some time later_
>Period n+x: We add more tokens, purchasing is possible again
>Period n+x+1: Sale continues
>
>Does the price update in period n+x, based on tokens sold in period n before we sold out, or does the price only update in period n+x+1 based on tokens sold in period n+x
>
>(for reference, the bug being fixed is that it would only update in period n+x+1, but do so based on tokens sold in period n and period n+x, which combined could only sell the amount expected to sell in one period)

**Arren**
>So, does the batch time start from 0 when the tokens are added, or could this be half way through a batch period?
>
>If it sold above the threshold in n, then it makes sense that n+x would increase in price.
However, if it sold below the threshold in n due to selling out of tokens, then it doesn't make sense that it would decrease the price.
>
>So, I think the middle ground for this is that it increases in n+x+1

**Alex**
>>So, does the batch time start from 0 when the tokens are added, or could this be half way through a batch period?

>It could be halfway through for both the original and new implementations
>>So, I think the middle ground for this is that it increases in n+x+1

>Or decreases, based only on what was sold in n+x?

**Arren**
>Yes, sorry, exactly